rct229/ruleset_functions/get_primary_secondary_loops_dict.py

```diff
@@ -0,0 +1,90 @@
+from rct229.data.schema_enums import schema_enums
+from rct229.ruleset_functions.baseline_systems.baseline_system_util import (
+    HVAC_SYS,
+    find_exactly_one_fluid_loop,
+)
+from rct229.ruleset_functions.get_baseline_system_types import get_baseline_system_types
+from rct229.utils.assertions import getattr_
+from rct229.utils.jsonpath_utils import find_all
+
+FLUID_LOOP = schema_enums["FluidLoopOptions"]
+
+APPLICABLE_SYS_TYPES = [
+    HVAC_SYS.SYS_7,
+    HVAC_SYS.SYS_8,
+    HVAC_SYS.SYS_11_1,
+    HVAC_SYS.SYS_11_2,
+    HVAC_SYS.SYS_12,
+    HVAC_SYS.SYS_13,
+    HVAC_SYS.SYS_7B,
+    HVAC_SYS.SYS_8B,
+    HVAC_SYS.SYS_11_1B,
+    HVAC_SYS.SYS_12B,
+]
+
+
+def get_primary_secondary_loops_dict(rmi_b):
+    """
+    Get the list of primary and secondary loops for CHW for a B-RMI.
+
+    Parameters
+    ----------
+    rmi_b: A baseline ruleset model instance
+
+    Returns: primary_secondary_loops_dict
+    A dictionary that saves pairs of primary and secondary loops for
+    baseline chilled water system, e.g. {primary_loop_1.id: [secondary_loop_1.id,
+    secondary_loop2.id], primary_loop_2.id: [secondary_loop3.id]]}. If B-RMI does
+    not have primary-secondary loop configuration setup, return an empty dictionary.
+    """
+    baseline_hvac_system_dict = get_baseline_system_types(rmi_b)
+
+    # Use find_all here to avoid including None for missing cooling_loops
+    # Note: a chiller_loop id could appear more than once in this list
+    chiller_loop_ids = find_all("$.chillers[*].cooling_loop", rmi_b)
+
+    applicable_hvac_ids = [
+        hvac_id
+        for sys_type in APPLICABLE_SYS_TYPES
+        for hvac_id in baseline_hvac_system_dict[sys_type]
+    ]
+    applicable_hvac_systems = [
+        hvac
+        for hvac in find_all(
+            "$..building_segments[*].heating_ventilation_air_conditioning_systems[*]",
+            rmi_b,
+        )
+        if hvac["id"] in applicable_hvac_ids
+    ]
+    non_process_chw_coil_loop_ids = [
+        # Get hvac["cooling_system"]["chilled_water_loop"] or raise exception
+        getattr_(hvac, "hvac system", "cooling_system", "chilled_water_loop")
+        for hvac in applicable_hvac_systems
+    ]
+    # Initialize variables
+    primary_loops = []
+    tmp_primary_secondary_loops_dict = dict()
+
+    # Iterate through cooling type fluid loops
+    for chilled_fluid_loop in find_all(
+        f'fluid_loops[*][?(@.type="{FLUID_LOOP.COOLING}")]', rmi_b
+    ):
+        cfl_id = chilled_fluid_loop["id"]
+        if cfl_id in chiller_loop_ids and cfl_id in non_process_chw_coil_loop_ids:
+            # No loop in baseline shall be
+            tmp_primary_secondary_loops_dict = dict()
```

Comment:
Could you check if this line can be deleted since nothing happens after being defined in line 66?